File: Habitica/src/main/java/com/habitrpg/android/habitica/models/Tag.java
package com.habitrpg.android.habitica.models;

import com.habitrpg.android.habitica.models.tasks.Task;
import com.habitrpg.android.habitica.models.tasks.TaskTag;
import com.habitrpg.android.habitica.models.user.User;

import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Description of a Tag in HabitRPG
 * Created by MagicMicky on 16/03/14.
 */

public class Tag extends RealmObject {

    @PrimaryKey
    public String id;

    public User user;
    public RealmList<Task> tasks;
    String name;
    boolean challenge;

    public List<Task> getTasks() {
        return tasks;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }


    @Override
    public boolean equals(Object o) {
        if (o.getClass().equals(Tag.class)) {
            Tag otherTag = (Tag) o;
            return this.getId().equals(otherTag.getId());
        }
        return super.equals(o);
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/Tag.kt
package com.habitrpg.android.habitica.models

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Tag : RealmObject(), BaseObject {

    @PrimaryKey
    var id: String = ""

    var userId: String? = null
    var name: String = ""
    var group: String? = null
    internal var challenge: Boolean = false

    override fun equals(other: Any?): Boolean {
        if (other is Tag) {
            return this.id == other.id
        }
        return super.equals(other)
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}
